db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -1177,7 +1177,9 @@ Status DBImpl::CompactRangeInternal(const CompactRangeOptions& options,
           break;
         }
         if (output_level == ColumnFamilyData::kCompactToBaseLevel) {
-          final_output_level = cfd->NumberLevels() - 1;
+          assert(cfd->current() && cfd->current()->storage_info());
+          VersionStorageInfo* vstorage = cfd->current()->storage_info();
```

Comment:
Do we need mutex_ or referenced super version for this? I'm not sure if this is thread safe.

Reply:
One other concern I realize now is if some other compaction that changed the based level that ran after `RunManualCompaction()` but before we get base_level here.